src/store/operations/illustration-remove.js

```diff
@@ -0,0 +1,30 @@
+import { projects } from '../init';
+
+/**
+ * Removes an illustration from a project.
+ * @param {Object} opts
+ * @param {String} opts.projectSlug Kebab-case project slug name
+ * @param {String} opts.illustrationSlug Kebe-case illustration slug name
+ * @returns
+ */
+export async function removeIllustration({ projectSlug, illustrationSlug }) {
+  const projectEntry = await projects.get(projectSlug);
+
+  if (projectEntry) {
+    const { illustrations } = projectEntry;
+    const illosUpdated = illustrations.filter(
+      (d) => d.slug !== illustrationSlug,
+    );
+
+    await projects.set(projectSlug, {
+      ...projectEntry,
+      illustrations: illosUpdated,
+    });
+
+    await projects.save();
+
+    return projects.get(projectSlug);
+  }
+
+  throw new Error(`No such project exists for: ${projectSlug}`);
```

Comment:
Generally it avoids extra nesting if you do the checks and throw the error before you continue running the function. i.e.

```
if(!projectEntry){
  throw new Error(`No such project exists for: ${projectSlug}`);
}

...
```

There's a name for this principle somewhere but don't feel like digging lol.